core/services/ocr2/plugins/ocr2keeper/evm21/transmit/cache_test.go

```diff
@@ -0,0 +1,82 @@
+package transmit
+
+import (
+	"testing"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg/v3/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTransmitEventCache_Sanity(t *testing.T) {
+	tests := []struct {
+		name        string
+		cap         int64
+		logIDsToAdd []string
+		eventsToAdd []ocr2keepers.TransmitEvent
+		toGet       []string
+		expected    map[string]ocr2keepers.TransmitEvent
+	}{
+		{
+			"empty cache",
+			10,
+			[]string{},
+			[]ocr2keepers.TransmitEvent{},
+			[]string{"1"},
+			map[string]ocr2keepers.TransmitEvent{},
+		},
+		{
+			"happy path",
+			10,
+			[]string{"1", "2", "3", "4"},
+			[]ocr2keepers.TransmitEvent{
+				{WorkID: "1", TransmitBlock: 1},
+				{WorkID: "2", TransmitBlock: 2},
+				{WorkID: "3", TransmitBlock: 3},
+				{WorkID: "4", TransmitBlock: 4},
+			},
+			[]string{"1", "3"},
+			map[string]ocr2keepers.TransmitEvent{
+				"1": {WorkID: "1", TransmitBlock: 1},
+				"3": {WorkID: "3", TransmitBlock: 3},
+			},
+		},
+		{
+			"overflow",
+			3,
+			[]string{"1", "2", "3", "4", "5"},
+			[]ocr2keepers.TransmitEvent{
+				{WorkID: "1", TransmitBlock: 1},
+				{WorkID: "2", TransmitBlock: 2},
+				{WorkID: "3", TransmitBlock: 3},
+				{WorkID: "4", TransmitBlock: 4},
+				{WorkID: "5", TransmitBlock: 5},
```

Comment:
i think this test passed right now with the block number bug since logs were added in order. If you reverse the order here then the test will become more robust